core/node/proof_data_handler/src/request_processor.rs

```diff
@@ -226,58 +226,64 @@ impl RequestProcessor {
                     .unwrap()
                     .expect("Proved block without metadata");
 
-                let is_pre_boojum = l1_batch
+                let protocol_version = l1_batch
                     .header
                     .protocol_version
-                    .map(|v| v.is_pre_boojum())
-                    .unwrap_or(true);
-                if !is_pre_boojum {
-                    let events_queue_state = l1_batch
-                        .metadata
-                        .events_queue_commitment
-                        .expect("No events_queue_commitment");
-                    let bootloader_heap_initial_content = l1_batch
-                        .metadata
-                        .bootloader_initial_content_commitment
-                        .expect("No bootloader_initial_content_commitment");
-
-                    if events_queue_state != events_queue_state_from_prover
-                        || bootloader_heap_initial_content
-                            != bootloader_heap_initial_content_from_prover
-                    {
-                        let server_values = format!("events_queue_state = {events_queue_state}, bootloader_heap_initial_content = {bootloader_heap_initial_content}");
-                        let prover_values = format!("events_queue_state = {events_queue_state_from_prover}, bootloader_heap_initial_content = {bootloader_heap_initial_content_from_prover}");
-                        panic!(
-                            "Auxilary output doesn't match, server values: {} prover values: {}",
-                            server_values, prover_values
-                        );
-                    }
+                    .unwrap_or_else(ProtocolVersionId::last_potentially_undefined);
+
+                let events_queue_state = l1_batch
+                    .metadata
+                    .events_queue_commitment
+                    .expect("No events_queue_commitment");
+                let bootloader_heap_initial_content = l1_batch
+                    .metadata
+                    .bootloader_initial_content_commitment
+                    .expect("No bootloader_initial_content_commitment");
+
+                if events_queue_state != events_queue_state_from_prover
+                    || bootloader_heap_initial_content
+                        != bootloader_heap_initial_content_from_prover
+                {
+                    let server_values = format!("events_queue_state = {events_queue_state}, bootloader_heap_initial_content = {bootloader_heap_initial_content}");
+                    let prover_values = format!("events_queue_state = {events_queue_state_from_prover}, bootloader_heap_initial_content = {bootloader_heap_initial_content_from_prover}");
```

Comment:
Nit: I get that this is copy-pasted, but I don't understand the reasoning behind `format!` if it's only used inside another interpolated string.